### PR TITLE
feat: vllm llama integration

### PR DESCRIPTION
**Pull request by jorgeantonio21**
# Integrate Llama Model for Fast, Batched Inference in atoma-vllm

## Motivation

The atoma-vllm library aims to provide efficient, batched inference capabilities for large language models. Integrating the Llama model into our system will expand our support for popular and powerful language models, enabling users to leverage Llama's capabilities within our high-performance inference framework.

## Description

This PR introduces Llama model support to the atoma-vllm library, implementing the necessary components for model loading, execution, and integration with our existing infrastructure. Key changes include:

1. Added a new `llama.rs` file in the `models` directory to implement Llama-specific functionality.
2. Implemented the `ModelLoader`, `ModelMetadata`, and `ModelExecutor` traits for the `LlamaModel` struct.
3. Updated the `ModelFilePaths` struct to accommodate Llama's file structure.
4. Modified the `fetch` and `load` functions to handle Llama model files and configurations.
5. Integrated Llama-specific parameters and configurations into the existing model execution pipeline.
6. Added a test file `llama.rs` in the `tests` directory to ensure proper functionality of the Llama integration.
7. Updated dependencies in `Cargo.toml` to include necessary Llama-related libraries.
8. Made adjustments to existing code to accommodate Llama's specific requirements and ensure compatibility with our batched inference system.

These changes allow users to easily load and use Llama models within the atoma-vllm framework, benefiting from our optimized, batched inference capabilities.

## Breaking Changes

- The `ModelLoader` trait's `fetch` and `load` functions now have slightly different signatures to accommodate the new `ModelFilePaths` struct and additional parameters.
- The `ModelMetadata` trait has been updated with new method names and signatures. Any custom model implementations will need to be updated to match the new trait definition.

These breaking changes are necessary to support a wider range of models and improve overall system flexibility. Users of the library may need to update their code if they have implemented custom models or are directly interacting with low-level components of the system.

### CODE REVIEW

File: atoma-vllm/src/llm_service.rs [hunk 1]
```diff
@@ -45,40 +61,69 @@ pub struct LlmService {
...
-        tokenizer: Tokenizer,
+        tokenizer_receiver: mpsc::UnboundedReceiver<EncodeTokenizerRequest>,
         validation_service: Validation,
+        shutdown_signal: mpsc::Receiver<()>,
     ) -> Result<Self, LlmServiceError>
     where
         M: ModelExecutor + Send + Sync + 'static,
     {
+        let span = info_span!("llm-service");
+        let _span = span.clone();
```
Comment (Cifko) [inline atoma-vllm/src/llm_service.rs:97]: is this clone necessary?

File: atoma-vllm/src/worker.rs [hunk 2]
```diff
@@ -59,29 +60,28 @@ where
...
         dtype: DType,
-        model_name: String,
-        revision: String,
+        model: M,
         num_cpu_blocks: usize,
         num_gpu_blocks: usize,
         enable_chunked_prefill: bool,
     ) -> Result<Self, ModelWorkerError> {
+        let span = info_span!("model-worker");
+        let _span = span.clone();
```
Comment (Cifko) [inline atoma-vllm/src/worker.rs:73]: Is this clone necessary?

File: atoma-vllm/src/token_output_stream.rs
Comment (Cifko) [inline atoma-vllm/src/token_output_stream.rs:1]: This file is completely commented out

**Review by Cifko — APPROVED** [review]

File: atoma-vllm/src/llm_service.rs (hunk 1 above)
Comment (jorgeantonio21) [reply atoma-vllm/src/llm_service.rs:93]: Yeah, I can't make the code compile without the clone as the `span` gets borrowed. I can't either use 

`let _enter = span.clone().enter();`

As it frees the value while in use.

File: atoma-vllm/src/token_output_stream.rs
Comment (jorgeantonio21) [reply atoma-vllm/src/token_output_stream.rs:1]: Nice catch, removed !

**Pull request merged**